src/eval-frame/actions/actions.js

```diff
@@ -216,124 +209,96 @@ export function evaluateText(
-      }
-      dispatch(updateCellProperties(cell.id, cellProperties))
-      // dispatch(incrementExecutionNumber())
-      dispatch(appendToEvalHistory(cell.id, cell.content, output))
-      dispatch(updateUserVariables())
-    }
-
-    const messageCallback = (msg) => {
-      dispatch(appendToEvalHistory(cell.id, msg, undefined, { historyType: 'CELL_EVAL_INFO' }))
+      evaluationQueue = evaluationQueue.then(() =>
+        dispatch(evaluateCode(evalText, evalType, state)))
     }
-
-    return ensureLanguageAvailable(cell.language, cell, state, dispatch)
-      .then(language => runCodeWithLanguage(language, code, messageCallback))
-      .then(
-        output => updateCellAfterEvaluation(output),
-        output => updateCellAfterEvaluation(output, 'ERROR'),
-      )
-      .then(waitForExplicitContinuationStatusResolution)
-      .then(() => dispatch(temporarilySaveRunningCellID(undefined)));
-  }
-}
-
-function evaluateMarkdownCell(cell) {
-  return dispatch => dispatch(updateCellProperties(
-    cell.id,
-    {
-      value: MD.render(cell.content),
-      rendered: true,
-      evalStatus: 'SUCCESS',
-    },
-  ))
-}
-
-function evaluateCSSCell(cell) {
-  return (dispatch) => {
-    dispatch(updateCellProperties(
-      cell.id,
-      {
-        value: cell.content,
-        rendered: true,
-        evalStatus: 'SUCCESS',
-      },
-    ))
-    dispatch(appendToEvalHistory(
-      cell.id,
-      cell.content,
-      'Page styles updated',
-      { historyType: 'CELL_EVAL_INFO' },
-    ))
+    return evaluationQueue
   }
 }
 
-export function evaluateCell(cellId) {
-  return (dispatch, getState) => {
-    let cell
-    if (cellId === undefined) {
-      cell = getSelectedCell(getState())
-    } else {
-      cell = getCellById(getState().cells, cellId)
-    }
-
-    if (!cell.content) {
-      // if the cell has no content, evaluation is a no-op
-      return undefined
-    }
-
-    let evaluation
-    dispatch(incrementExecutionNumber())
-    // here is where we should mark a cell as PENDING.
-    if (cell.cellType === 'code') {
-      evaluationQueue = evaluationQueue
-        .then(() => dispatch(evaluateCodeCell(cell)))
-      evaluation = evaluationQueue
-    } else if (cell.cellType === 'markdown') {
-      evaluation = dispatch(evaluateMarkdownCell(cell))
-    } else if (cell.cellType === 'css') {
-      evaluation = dispatch(evaluateCSSCell(cell))
-    } else if (cell.cellType === 'fetch') {
-      evaluationQueue = evaluationQueue
-        .then(() => dispatch(evaluateFetchCell(cell)))
-      evaluation = evaluationQueue
-    } else if (cell.cellType === 'plugin') {
-      if (JSON.parse(cell.content).pluginType === 'language') {
-        evaluationQueue = evaluationQueue.then(() => dispatch(evaluateLanguagePluginCell(cell)))
-        evaluation = evaluationQueue
-      }
-    } else {
-      cell.rendered = false
-    }
-    return evaluation
-  }
-}
+// function evaluateCodeCell(cell) {
```

Comment:
can we cut this commented block? (np if you think we'll need it again, but it looks like it will probably go?)